src/App.tsx

```diff
@@ -1,35 +1,39 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
 import './App.css'
+import cronstrue from 'cronstrue'
 
 function App() {
-  const [count, setCount] = useState(0)
+  function generateCronExpression() {
+    function getRandomOrWildcard(min: number, max: number) {
+      if (Math.random() < 0.7) {
+        return '*';
+      } else {
+        return getRandomIntInclusive(min, max);
+      }
+    }
+    const minute = getRandomOrWildcard(0, 59);
+    const hour = getRandomOrWildcard(0, 23);
+    const dayOfMonth = getRandomOrWildcard(1, 31);
+    const month = getRandomOrWildcard(1, 12);
+    const dayOfWeek = getRandomOrWildcard(0, 6);
+
+    return `${minute} ${hour} ${dayOfMonth} ${month} ${dayOfWeek}`
+  }
+
+  const cronExpression = generateCronExpression();
 
   return (
     <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
+      <h1>crontab.training</h1>
+      <p>{cronExpression}</p>
+      <p>{cronstrue.toString(cronExpression)}</p>
     </>
   )
 }
 
+function getRandomIntInclusive(min: number, max: number) {
```

Comment:
Can we use [crypto.randomInt](https://nodejs.org/api/crypto.html#cryptorandomintmin-max-callback) instead?

Also, move the generator function to a different file, `generator.ts`.

Reply:
> Can we use [crypto.randomInt](https://nodejs.org/api/crypto.html#cryptorandomintmin-max-callback) instead?
Getting the following error: `Uncaught Error: Module "crypto" has been externalized for browser compatibility. Cannot access "crypto.randomInt" in client code.`

Relevant documentation: https://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility

How should we proceed?

Reply:
Gotcha, this is a nodejs only package. Let's stick to your implementation for now. The end goal is to move generation and validation to a backend server anyway.